File: PAT_Basic_Level_by_lua/1071 小赌怡情/1071 小赌怡情.lua
-- 题目编号: 1071 小赌怡情
-- 实现原理: 模拟赌博游戏过程，每轮读取玩家下注和猜的大小结果，判断玩家是否猜对并更新筹码。若筹码归零则游戏结束。
-- money: 初始筹码，n: 游戏轮数
local money, n = io.read("*n"), io.read("*n")
-- 逐轮游戏
for i = 1, n do
    -- 输入顺序为 n1、猜测方向、下注数、n2
    local a, bet, c, d = io.read("*n"), io.read("*n"), io.read("*n"), io.read("*n")
    if c > money then
        -- 筹码不足，无法下注
        print("Not enough tokens.  Total = " .. money .. ".")
    elseif (d > a) == (bet == 1) then
        -- 猜对了：a>b且猜大，或a<=b且猜小（即猜大不成立）
        money = money + c
        print("Win " .. c .. "!  Total = " .. money .. ".")
    else
        -- 猜错了
        money = money - c
        print("Lose " .. c .. ".  Total = " .. money .. ".")
    end
    -- 筹码归零，游戏结束
    if money == 0 then
        print("Game Over.")
        break
    end
end

File: PAT_Basic_Level_by_c/1071 小赌怡情/1071 小赌怡情.c
/*
 * 题目：1071 小赌怡情
 * 实现原理：
 *   简单的数字比较赌博游戏。
 *   规则：
 *   - b=0表示押小（n1 > n2），b=1表示押大（n1 < n2）
 *   - t是下注筹码数
 *   - 赢了获得t筹码，输了失去t筹码
 *   - 筹码为0时游戏结束
 *   算法步骤：
 *   1. 读取初始筹码T和游戏次数K
 *   2. 逐次读取每局游戏的n1, b, t, n2
 *   3. 根据规则判断输赢并更新筹码
 *   4. 筹码为0时立即结束游戏
 */
#include <stdio.h>

int main() {
    int T, K;  // T: 初始筹码; K: 游戏次数
    scanf("%d %d", &T, &K);
    for (int i = 0; i < K; i++) {
        int n1, b, t, n2;  // 两个数字 n1、n2，押注方向 b，筹码 t
        scanf("%d %d %d %d", &n1, &b, &t, &n2);
        if (T == 0) {  // 筹码为 0：游戏结束
            printf("Game Over.\n");
            break;
        }
        if (t > T) {  // 下注超过当前筹码：提示不足并跳过本局
            printf("Not enough tokens. Total = %d.\n", T);
            continue;
        }
        if ((n1 > n2 && b == 0) || (n1 < n2 && b == 1)) {  // 押小(b=0)且 n1>n2 或押大(b=1)且 n1<n2：赢
            T += t;
            printf("Win %d! Total = %d.\n", t, T);
        } else {  // 猜错：扣除筹码
            T -= t;
            printf("Lose %d. Total = %d.\n", t, T);
            if (T == 0) {  // 输光筹码：游戏结束
                printf("Game Over.\n");
                break;
            }
        }
    }
    return 0;
}